pkg/api/validation/validation_test.go

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2022 by the OnMetal authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package validation
+
+import (
+	"github.com/onmetal/machine-controller-manager-provider-onmetal/pkg/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/types"
+)
+
+var fldPath *field.Path
+
+var _ = Describe("Machine", func() {
+
+	DescribeTable("ValidateProviderSpecAndSecret",
+		func(spec *v1alpha1.ProviderSpec, secret *corev1.Secret, fldPath *field.Path, match types.GomegaMatcher) {
+			errList := ValidateProviderSpecAndSecret(spec, secret, fldPath)
+			Expect(errList).To(match)
+		},
+		Entry("no secret",
+			&v1alpha1.ProviderSpec{
+				RootDisk: &v1alpha1.RootDisk{},
+			},
+			nil,
+			fldPath,
+			Not(ContainElement(field.Required(fldPath.Child(""), "corev1 secret required"))),
+		),
+		Entry("no userData in secret",
+			&v1alpha1.ProviderSpec{
+				RootDisk: &v1alpha1.RootDisk{},
+			},
+			&corev1.Secret{
+				Data: map[string][]byte{
+					"userData": nil,
+				},
+			},
+			fldPath,
+			Not(ContainElement(field.Required(fldPath.Child("userData"), "corev1 secret userData required"))),
+		),
+		Entry("no kubeconfig in secret",
```

Comment:
The `kubeconfig` in the secret will go. Please remove this check.

Reply:
Addressed.